core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -733,4 +803,9 @@ void assertEquals(String context, Object expected, Object actual) {
   protected interface Action {
     void invoke();
   }
+
+  @FunctionalInterface
+  interface CheckedFunction<T, R> {
```

Comment:
This doesn't seem necessary. Can you just use `Function` and swallow the IOException?

Reply:
Removed, as I refactored the methods that used it, so it's no longer needed.